### ISSUE TITLE
I managed to Rotate elements using mouse but i need help to improve the code

### ISSUE THREAD

**Issue by jloguercio** [taye/interact.js#920]
### Actual behavior
I managed to make the Rotatable element along with the Draggable and the Resizable using the mouse in a PC following an example I found here, i seems works fine but the element rotations feels weird, here is a video

https://user-images.githubusercontent.com/12191804/120106010-f9438c80-c120-11eb-823b-422598a68259.mp4

And here is my code :

                    
      interact('.element')
            .resizable({
                allowFrom: '.dg-controls',
                edges: {
                    top: true,
                    bottom: true,
                    left: true,
                    right: true
                },
                modifiers: [
                    // keep the edges inside the parent
                    interact.modifiers.restrictEdges({
                        outer: 'parent'
                    }),
                    interact.modifiers.restrictSize({
                        min: {width: 50, height: 50}
                    })
                ],
                listeners: {
                    start(event) {
                        if (event._interaction.downEvent.path[0].classList.contains('dg-rotator')) {
                            const element = event.target;
                            const rect = element.getBoundingClientRect();

                            // store the center as the element has css `transform-origin: center center`
                            element.dataset.centerX = rect.left + rect.width / 2;
                            element.dataset.centerY = rect.top + rect.height / 2;
                            // get the angle of the element when the drag starts
                            element.dataset.angle = getDragAngle(event);
                        }
                    },
                    move(event) {
                        let x = (parseFloat(event.target.getAttribute('data-x')) || 0)
                        let y = (parseFloat(event.target.getAttribute('data-y')) || 0)
                        let angle = (parseFloat(event.target.getAttribute('data-angle')) || 0);

                        if (event._interaction.downEvent.path[0].classList.contains('dg-rotator')) {
                            angle = getDragAngle(event);
                        } else {
                            // update the element's style
                            event.target.style.width = event.rect.width + 'px'
                            event.target.style.height = event.rect.height + 'px'

                            // translate when resizing from top or left edges
                            x += event.deltaRect.left
                            y += event.deltaRect.top

                            event.target.setAttribute('data-x', x)
                            event.target.setAttribute('data-y', y)
                        }

                        event.target.style.transform =
                            'translate(' + event.target.getAttribute('data-x') + 'px,' + event.target.getAttribute('data-y') + 'px) rotate(' + angle + 'rad' + ')';
                    },
                    end(event) {
                        if (event._interaction.downEvent.path[0].classList.contains('dg-rotator')) {
                            event.target.dataset.angle = getDragAngle(event);
                        }
                    }
                }
            })
            .draggable({
                modifiers: [
                    interact.modifiers.restrictRect({
                        restriction: 'parent',
                        endOnly: true
                    })
                ],
                listeners: {
                    move(event) {
                        event.target.setAttribute('data-x', (parseFloat(event.target.getAttribute('data-x')) || 0) + event.dx)
                        event.target.setAttribute('data-y', (parseFloat(event.target.getAttribute('data-y')) || 0) + event.dy)
                        event.target.setAttribute('data-angle', parseFloat(event.target.dataset.angle) || 0);
                        event.target.style.transform =
                            'translate(' + event.target.getAttribute('data-x') + 'px, ' + event.target.getAttribute('data-y') + 'px) rotate(' + event.target.getAttribute('data-angle') + 'rad' + ')';
                    }
                },
            })
            
            function getDragAngle(event) {
            var element = event.target;
            var startAngle = parseFloat(element.dataset.angle) || 0;
            var center = {
                x: parseFloat(element.dataset.centerX) || 0,
                y: parseFloat(element.dataset.centerY) || 0,
            };
            var angle = Math.atan2(center.y - event.clientY,
                center.x - event.clientX);

            return angle - startAngle;
        }

Basically i use the allowFrom from resizable to handle the rotatable using the element that fire the event (in this case the class dg-rotator)

I would appreciate your help with improving the rotation code, thank you very much

### System configuration
**interact.js version**:  1.10.11
**Browser name and version**: Chrome 
**Operating System**: Ubuntu 20.04

**Comment by LukeTOBrien** [comment]
I would recommend you to ask this question [on Stackoverflow](https://stackoverflow.com/questions/tagged/interact.js?tab=Newest), you might get more replies ;-)
